packages/swap/src/pricing/index.ts

```diff
@@ -1,58 +1,97 @@
 import axios from 'axios';
 import { InternalAsset } from '@/shared/enums';
 import env from '../config/env';
-import { CacheMap } from '../utils/dataStructures';
 import logger from '../utils/logger';
 import { deferredPromise } from '../utils/promise';
 
 const COINGECKO_VS_CURRENCY = 'usd';
 
 // https://api.coingecko.com/api/v3/coins/list?include_platform=true
-const coinGeckoIdMap: Record<InternalAsset, string> = {
+export const coinGeckoIdMap = {
   Flip: 'chainflip',
   Usdc: 'usd-coin',
   Dot: 'polkadot',
   Eth: 'ethereum',
   Btc: 'bitcoin',
   Usdt: 'tether',
-};
-const priceCache = new CacheMap<string, Promise<number | undefined>>(30_000);
+} as const satisfies Record<InternalAsset, string>;
+
+type CoingeckoId = (typeof coinGeckoIdMap)[InternalAsset];
+
+type CoingeckoPriceResponse = Record<CoingeckoId, Record<typeof COINGECKO_VS_CURRENCY, number>>;
 
 const coingeckoAxios = axios.create({
   baseURL: 'https://pro-api.coingecko.com/api/v3',
   timeout: 5000,
   headers: { 'x-cg-pro-api-key': env.COINGECKO_API_KEY },
 });
 
-export const getAssetPrice = async (asset: InternalAsset): Promise<number | undefined> => {
-  logger.debug(`getting asset price for "${asset}"`);
+export class PriceCache {
```

Comment:
using a class for encapsulation is easier than trying to reset global variables by resetting mocks in jest